contrib/demo/prototype3-script/script-apibindings.sh

```diff
@@ -0,0 +1,140 @@
+pe "kubectl apply -f ${DEMO_DIR}/apibinding/cert-manager-apibinding.yaml"
+
+c "The kcp system will do the binding now in the background and make the APIs available:"
+pe "kubectl get apibinding -o yaml"
+
+sleep 2
+
+pe "kubectl get apibinding -o yaml"
+c "We see it is bound and ready. We should now see the Certificate API available 🤞:"
+pe "kubectl api-resources"
+pe "kubectl get certificates"
+
+c "Note that this is much more than just a CRD: we have bound to the whole cert-manager service. So certificate requests will actually work 🎉"
+
+#fi
+
+c "Let's start by creating a private key for ca issuer. This step requires openssl to be installed locally"
+pe "openssl req -nodes -x509 -newkey rsa:2048 \
+  -subj \"/C=US/ST=New York/L=New York City/O=Kubernetes/OU=KCP/CN=example.com/emailAddress=test@example.com\" \
+  -keyout ${KCP_DATA_DIR}/ca.key \
+  -out ${KCP_DATA_DIR}/ca.crt \
+  -extensions v3_ca -config ${DEMO_DIR}/apibinding/openssl.cnf"
+kubectl delete issuers ca-issuer &>/dev/null
+kubectl delete certificates serving-cert &>/dev/null
+kubectl delete secret ca-key-pair serving-cert &>/dev/null
+pe "kubectl create secret tls ca-key-pair --cert=${KCP_DATA_DIR}/ca.crt --key=${KCP_DATA_DIR}/ca.key"
+
+c "Let's request a serving cert for our web application:"
+
+pe "cat ${DEMO_DIR}/apibinding/issuer.yaml"
+pe "cat ${DEMO_DIR}/apibinding/certificate.yaml"
+
+pe "kubectl apply -f ${DEMO_DIR}/apibinding/issuer.yaml"
+pe "kubectl apply -f ${DEMO_DIR}/apibinding/certificate.yaml"
+pe "kubectl get secrets"
+
+c "Now let's deploy an app. First we need a target workload cluster."
+cat <<EOF > "${KCP_DATA_DIR}/cluster-us-east1.yaml"
+apiVersion: workload.kcp.dev/v1alpha1
+kind: WorkloadCluster
+metadata:
+  name: kind-us-east1
+spec:
+  kubeconfig: |
+$(sed 's,^,    ,' "${CLUSTERS_DIR}"/us-east1.kubeconfig)
+EOF
+pe "kubectl apply -f ${KCP_DATA_DIR}/cluster-us-east1.yaml"
+
+c "Let's wait for kcp to have the cluster syncing ready"
+pe "kubectl wait --for condition=Ready workloadcluster/kind-us-east1"
+
+wait
+clear
+
+# Use the right arch for the kuard image
+kuardArch=$(uname -m)
+if [[ "${kuardArch}" == "x86_64" ]]; then
+  kuardArch=amd64
+fi
+sed "s/ARCH/$kuardArch/" "${DEMO_DIR}"/apibinding/deployment-kuard-tls.yaml > "${KCP_DATA_DIR}"/deployment-kuard-tls.yaml
+pe "cat ${KCP_DATA_DIR}/deployment-kuard-tls.yaml"
+pe "kubectl apply -f ${KCP_DATA_DIR}/deployment-kuard-tls.yaml"
+
+c "Take note of Replicas:"
+pe "kubectl describe deployment/kuard"
+
+c "Let's wait for the deployment to be running in kind"
+pe "while ! kubectl --kubeconfig ${CLUSTERS_DIR}/us-east1.kubeconfig wait --for=jsonpath='{.status.availableReplicas}'=1 -A -l 'workloads.kcp.dev/cluster=kind-us-east1' --field-selector 'metadata.name=kuard' deployments; do
+  sleep 1
+done"
+
+c "Let's describe it to see its status - take note of Replicas"
+pe "kubectl describe deployment/kuard"
+
+c "Does it actually serve with cert-manager-produced certs? We will find out."
+pe ""
```

Comment:
The last few steps are:
1. Applying this service:
```yaml
apiVersion: v1
kind: Service
metadata:
  name: kuard
spec:
  selector:
    app: kuard
  ports:
    - protocol: TCP
      port: 9376
      targetPort: 8080
```

2. Make sure the service is created:
```
pe "kubectl get svc"
```

3. Port forward to expose the web app
```
kubectl port-forward service/kuard 9376:9376
``` 

4. `curl --cert ${KCP_DATA_DIR}/ca.crt --key ${KCP_DATA_DIR}/ca.key http://127.0.0.1:9376 -v`
We are able to reach the server. You should be able to see a `200 OK` in the verbose output to confirm :)

Reply:
Was a little harder. The port-forward didn't work because we don't know the namespace name on the workload cluster (finding out about it would be cheating).

The kcp service does not support port-forward yet.

So with @jmprusi's help (thank you!) I switched to nginx ingress, which is deployed anyway. After enabling ssl-passthrough in the nginx deployment, and after fixing the curl flags to use `--cacert` and `--resolve` to workaround my broken internet provider (not allowing 127.0.0.1 from nip.io) it started working 🎉